pipeline_dp/dp_engine.py

```diff
@@ -111,6 +113,37 @@ def _aggregate(self, col, params: AggregateParams,
 
         return col
 
+    def _check_aggregate_params(self, col, params: AggregateParams,
+                                data_extractors: DataExtractors):
+
+        if col is None or not col:
+            raise Exception("col must be non-empty")
+        if params is None or not isinstance(params, AggregateParams):
+            raise Exception("params must be set to valid AggregateParams")
+        if not isinstance(params.max_partitions_contributed,
+                          int) or params.max_partitions_contributed <= 0:
+            raise Exception("params.max_partitions_contributed must be set "
+                            "to a positive integer")
+        if not isinstance(params.max_contributions_per_partition,
+                          int) or params.max_contributions_per_partition <= 0:
+            raise Exception(
+                "params.max_contributions_per_partition must be set "
+                "to a positive integer")
+        needs_low_high = any(metric == Metrics.SUM for metric in params.metrics)
+        if needs_low_high and (params.low is None or params.high is None):
+            raise Exception("params.low and params.high must be set")
+        if needs_low_high and (math.isinf(params.low) or math.isnan(
+                params.low) or math.isinf(params.high) or
```

Comment:
Nit: extract math.isinf(...) and math.isnan(...) to a helper function somethig like `_is_proper_number`. you can put it in this file in the end, since this function is private, it's enough to have a 1-line description, i.e. without comments about arguments and returns.